vehicle/IEngine.h

```diff
@@ -0,0 +1,15 @@
+#ifndef IENGINE_H
+#define IENGINE_H
+
+#include <string>
+#include <memory> 
+
+// Engine interface
+class Engine {
```

Comment:
same here, I'd call it "IEngine" to be clear

Reply:
ok, changed